spec/features/monitoring_feature_spec.rb

```diff
@@ -10,6 +10,15 @@
     visit authenticated_root_path
   end
 
+  scenario "It works if there are no minions", js: true do
```

Comment:
I think test would be green even without the changes in the code. The problem exists when the message is printed once but then all the Minions get assigned somehow. In that case the messages was never deleted. To check the correct case you first need to visit the page, then (optionally) verify message existence, assign the Minions and verify message removal.